packages/vue/src/components/CheckoutButton.vue

```diff
@@ -0,0 +1,53 @@
+<script setup lang="ts">
+import { useAttrs, useSlots } from 'vue';
+import type { __experimental_CheckoutButtonProps } from '@clerk/types';
+import { useClerk } from '../composables/useClerk';
+import { useAuth } from '../composables/useAuth';
+import { assertSingleChild, normalizeWithDefaultValue } from '../utils';
+
+const props = defineProps<__experimental_CheckoutButtonProps>();
+
+const clerk = useClerk();
+const { userId, orgId } = useAuth();
+const slots = useSlots();
+const attrs = useAttrs();
+
+// Authentication checks - similar to React implementation
+if (userId.value === null) {
+  throw new Error('Ensure that `<CheckoutButton />` is rendered inside a `<SignedIn />` component.');
+}
+
+if (orgId.value === null && props.for === 'organization') {
+  throw new Error('Wrap `<CheckoutButton for="organization" />` with a check for an active organization.');
+}
```

Comment:
Should we also throw this error when value is still `undefined` (Clerk is loading)?